### PR TITLE
CI testing of supported dependency versions (Closes #366) (Closes #421)

### LINKED ISSUE #366: CI not using requested versions of numpy
#365 happened because our CI is happily installing our very restricted version of numpy, and then proceeding to install scipy which smashes the numpy version.

```
$ pip install --force-reinstall "numpy${NUMPY_VERSION}"
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting numpy<1.7.0,>=1.6.0
  Downloading numpy-1.6.2-cp27-cp27mu-manylinux1_x86_64.whl (13.6 MB)
Installing collected packages: numpy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.16.6
    Uninstalling numpy-1.16.6:
      Successfully uninstalled numpy-1.16.6
Successfully installed numpy-1.6.2
install.2
7.28s$ pip install scipy
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting scipy
  Downloading scipy-1.2.3-cp27-cp27mu-manylinux1_x86_64.whl (24.8 MB)
Collecting numpy>=1.8.2
  Using cached numpy-1.16.6-cp27-cp27mu-manylinux1_x86_64.whl (17.0 MB)
Installing collected packages: numpy, scipy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.6.2
    Uninstalling numpy-1.6.2:
      Successfully uninstalled numpy-1.6.2
Successfully installed numpy-1.16.6 scipy-1.2.3
```

To continue testing old numpy versions we'll have to change that approach, probably involving either specifying versions of everything or making a single pip call for all installs.

### LINKED ISSUE #421: test_pybats.TestBats2d.testPlotting failure
A pybats test is failing (this is probably numpy version dependent).

### Minimal example to reproduce issue:

```
python3 test_pybats.py TestBats2d.testPlotting
```

### Error message/Traceback:

```
======================================================================
ERROR: testPlotting (__main__.TestBats2d)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_pybats.py", line 224, in testPlotting
    self.mhd.add_stream_scatter('ux','uz',target=ax,narrow=1)
  File "/home/jtniehof/.local//lib//python3.5/site-packages/spacepy/pybats/bats.py", line 1553, in add_stream_scatter
    method=method)
  File "/home/jtniehof/.local//lib//python3.5/site-packages/spacepy/pybats/bats.py", line 1355, in get_stream
    maxPoints=maxPoints, method=method, extract=extract)
  File "/home/jtniehof/.local//lib//python3.5/site-packages/spacepy/pybats/bats.py", line 455, in __init__
    self.treetrace(bats, maxPoints=maxPoints)
  File "/home/jtniehof/.local//lib//python3.5/site-packages/spacepy/pybats/bats.py", line 588, in treetrace
    for i in range(1, xbwd.size):
AttributeError: 'list' object has no attribute 'size'

----------------------------------------------------------------------
Ran 1 test in 0.229s

FAILED (errors=1)
```

### OS, Python version, and dependency version information:
```
Linux-4.4.0-98-generic-x86_64-with-Ubuntu-16.04-xenial
sys.version_info(major=3, minor=5, micro=2, releaselevel='final', serial=0)
numpy=1.11.0
scipy=0.17.0
matplotlib=1.5.1
```

### Version of SpacePy
Current master (22ceeef889d36cf9a7717206610c7cc814262c7a)

### PR DESCRIPTION
**Pull request by jtniehof**
This PR selects specific versions of all our dependencies so that we're explicitly testing the oldest supported version of everything. This includes rearranging how the pip installs are done so we get the requested version of numpy (Closes #366). By necessity this also gets the pybats tests working on matplotlib 1.5.1 (Closes #421). And to help make it all work there's a quick shell script that builds conda environments and runs the tests in them, basically mimicking CI so we don't have to open a PR to test against a bunch of versions. Oh and it fixes omni and omni-related tests when running the tests with a working directory other than the test directory.

## PR Checklist

- [X] Pull request has descriptive title
- [X] Pull request gives overview of changes
- [X] New code has inline comments where necessary
- [X] (N/A) Any new modules, functions or classes have docstrings consistent with SpacePy style
- [X] (N/A) Added an entry to CHANGELOG if fixing a major bug or providing a major new feature
- [X] New features and bug fixes should have unit tests
- [X] Relevant issues are linked to (e.g. `See issue #` or `Closes #`)

### CODE REVIEW

**Comment by jtniehof** [comment]
Okay, I'm a bit thrown by this one. AstroPy's bundled pyfits forces `-Wno-format` into `extra_compile_options` (`astropy/io/fits/setup_package.py` line 33) but this conflicts with `-Werror=format-security` which is set by *shrugs*. Not sure how to disable `-Wno-format` since it's explicitly added *after* all other stuff. See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=871156

I suppose maybe I could try `-Wno-error=format-security` to disable the security checks and see how that goes?

**Comment by jtniehof** [comment]
@spacecataz , asking for you for the pybats changes. @drsteve since you set up the CI and so you're the other stuckee with it.

**Review by drsteve — COMMENTED** [review]
CI changes look okay to me. Test logs on travis look good too. I'll let @spacecataz approve/merge once he's blessed the pybats changes.

**Review by spacecataz — APPROVED** [review]

**Pull request merged**
